[ffmpeg] Add support for dav1d,fontconfig,freetype,fribidi,ilbc,modplug,opengl,openjpeg,libssh,tensorflow,tesseract,webp,libxml2 dependencies.

Sibras opened the pull request:
* [ffmpeg] Add support for fontconfig,freetype,fribidi,modplug,openjpeg,libssh,tesseract,libxml2 dependencies.
* [ffmpeg] Add suport for libass,dav1d,libilbc, tensorflow.
* [tesseract] Use vcpkg_fixup_pkgconfig.
* [libxml2] Correct pkgconfig lib name.
* [libwebp] Use vcpkg_fixup_pkgconfig.
* [libssh] Export pkgconfig on windows.
* [modplug] Export pkgconfig on windows.
* [fribidi] Use vcpkg_fixup_pkgconfig.
* [openjpeg] Correct required static link libs in pkgconfig.
* [ffnvcodec] Set uwp as unsupported.
* [aubio] Fix building with updated FFmpeg.
* [pangolin] Fix building with updated FFmpeg.
* [libvpx] Enable arm-uwp builds.
* [fontconfig] Disable unused dependencies

Also added a convenience dependency to select all available dependencies in a single option "ffmpeg[all]". Also have ffmpeg[all-gpl] and ffmpeg[all-nonfree] as supersets of ffmpeg[all] with additional dependencies as allowed by the different licensing (ffmpeg[all] is default lgpl)

Some of the dependencies do not work on certain configurations so they are disabled where necessary.

Closes: #13227
Closes: #13111

Sibras committed:
[tesseract] Use vcpkg_fixup_pkgconfig.

Sibras committed:
[libxml2] Correct pkgconfig lib name.

Sibras committed:
[libwebp] Use vcpkg_fixup_pkgconfig.

Sibras committed:
[libssh] Export pkgconfig on windows.

Also move to using git to get source.

Sibras committed:
[modplug] Export pkgconfig on windows.

Sibras committed:
[ffmpeg] Add support for fontconfig,freetype,fribidi,modplug,openjpeg,libssh,tesseract,libxml2 dependencies.

Sibras committed:
[openjpeg] Correct required static link libs in pkgconfig.

cenit commented on ports/fribidi/portfile.cmake:
```diff
@@ -17,6 +17,7 @@ vcpkg_configure_meson(
 vcpkg_install_meson()
 vcpkg_fixup_pkgconfig()
 vcpkg_copy_pdbs()
+vcpkg_fixup_pkgconfig()
```
duplicate of two lines above? This port didn't require update

Sibras replied:
Yep, got an duplicated line at some point during a merge. Ive dropped the offending commit.

JackBoosY commented:
Ping @cenit for review again.

cenit commented on ports/libmodplug/portfile.cmake:
```diff
@@ -7,9 +7,10 @@ if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
-            "001-automagically-define-modplug-static.patch"
-            "002-detect_sinf.patch"
-            "003-use-static-cast-for-ctype.patch"
+            001-automagically-define-modplug-static.patch
```
Why not put this patch in a variable and write vcpkg_from_github only once?

Sibras replied:
done

cenit reviewed (changes requested):
Only a minor change maybe might be preferable

Sibras committed:
[modplug] Combine vcpkg_from_github using variable.

JackBoosY committed:
Merge branch 'master' into master

JackBoosY commented on ports/openjpeg/portfile.cmake:
```diff
@@ -37,8 +37,12 @@ vcpkg_configure_cmake(
 vcpkg_fixup_cmake_targets()
-if(VCPKG_TARGET_IS_WINDOWS)
-    # TODO: remove -lm from *.pc files
+if(WIN32 AND (NOT MINGW))
```
This seems does not work, can you confirm this?

JackBoosY replied:
```suggestion
if(VCPKG_TARGET_IS_WINDOWS AND (NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL MinGW))
```

JackBoosY commented on ports/libssh/CONTROL:
```diff
@@ -1,6 +1,6 @@
 Source: libssh
 Version: 0.9.5
-Port-Version: 1
+Port-Version: 3
```
```suggestion
Port-Version: 2
```

JackBoosY committed:
Update ports/libssh/CONTROL

Commit:
Improve portfile.cmake

Commit:
update version records.

Sibras committed:
[openjpeg] Update libs in pkgcfg.

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

Commit:
update version records

JackBoosY commented:
Need test all features.

rdstevens commented:
vcpkg install ffmpeg[all-gpl]:x64-windows-static

ERROR: libssh not found using pkg-config

JackBoosY commented:
When building `ffmpeg[ffmpeg,ffplay,ffprobe,avcodec,avformat,avdevice,avfilter,postproc,swresample,swscale,avresample,nonfree,gpl,version3,all-gpl,all-nonfree,bzip2,iconv,fdk-aac,fontconfig,freetype,fribidi,lzma,modplug,mp3lame,nvcodec,opencl,openjpeg,openssl,opus,sdl2,snappy,soxr,speex,ssh,tesseract,theora,vorbis,vpx,wavpack,webp,x264,xml2,zlib]:x64-windows-static`
```
=== CONFIGURING ===
ERROR: fontconfig not found using pkg-config

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
```

rdstevens commented:
When building:
```
.\vcpkg.exe install ffmpeg[ffmpeg,ffprobe,ffplay,gpl,x264,x265,bzip2,iconv,freetype,fribidi,lzma,mp3lame,nvcodec,opencl,openjpeg,opus,sdl2,snappy,soxr,theora,vorbis,vpx,wavpack,webp,zlib]:x64-windows  --recurse
```

CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:145 (message):
    Command failed: G:/code/sibras/vcpkg/downloads/tools/cmake-3.19.2-windows/cmake-3.19.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v -j13
    Working Directory: G:/code/sibras/vcpkg/buildtrees/wavpack/x64-windows-dbg
    See logs for more information:
      G:\code\sibras\vcpkg\buildtrees\wavpack\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:27 (vcpkg_build_cmake)
  ports/wavpack/portfile.cmake:25 (vcpkg_install_cmake)
  scripts/ports.cmake:128 (include)


Error: Building package wavpack:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: wavpack:x64-windows
  Vcpkg version: 2021-01-13-unknownhash

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5894317/install-x64-windows-dbg-out.log)

Sibras committed:
Merge branch 'master' into master

Sibras committed:
[libssh] Add pthread to pkgconfig when using mbedtls.

Sibras commented:
> 
> 
> [config.log](https://github.com/microsoft/vcpkg/files/5878985/config.log)
> C:\code\sibras\vcpkg>vcpkg install ffmpeg[all-gpl]:x64-windows-static
> 
> ERROR: libssh not found using pkg-config
> 
> C:\code\sibras\vcpkg>vcpkg install libssh:x64-windows-static
> Computing installation plan...
> The following packages are already installed:
> libssh[core,crypto,mbedtls]:x64-windows-static -> 0.9.5#2
> Package libssh:x64-windows-static is already installed
> 
> Total elapsed time: 377.6 us
> 
> The package libssh is compatible with built-in CMake targets:
> 
> ```
> find_package(libssh CONFIG REQUIRED)
> target_link_libraries(main PRIVATE ssh)
> ```

Fixed in [libssh] Add pthread to pkgconfig when using mbedtls.

Sibras commented:
> 
> 
> When building `ffmpeg[ffmpeg,ffplay,ffprobe,avcodec,avformat,avdevice,avfilter,postproc,swresample,swscale,avresample,nonfree,gpl,version3,all-gpl,all-nonfree,bzip2,iconv,fdk-aac,fontconfig,freetype,fribidi,lzma,modplug,mp3lame,nvcodec,opencl,openjpeg,openssl,opus,sdl2,snappy,soxr,speex,ssh,tesseract,theora,vorbis,vpx,wavpack,webp,x264,xml2,zlib]:x64-windows-static`
> 
> ```
> === CONFIGURING ===
> ERROR: fontconfig not found using pkg-config
> 
> If you think configure made a mistake, make sure you are using the latest
> version from Git.  If the latest version fails, report the problem to the
> ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
> Include the log file "ffbuild/config.log" produced by configure as this will help
> solve the problem.
> ```
> 
> [config.log](https://github.com/microsoft/vcpkg/files/5877467/config.log)

Fixed in ""

> 
> 
> When building:
> 
> ```
> .\vcpkg.exe install ffmpeg[ffmpeg,ffprobe,ffplay,gpl,x264,x265,bzip2,iconv,freetype,fribidi,lzma,mp3lame,nvcodec,opencl,openjpeg,opus,sdl2,snappy,soxr,theora,vorbis,vpx,wavpack,webp,zlib]:x64-windows  --recurse
> ```
> 
> CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:145 (message):
> Command failed: G:/code/sibras/vcpkg/downloads/tools/cmake-3.19.2-windows/cmake-3.19.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v -j13
> Working Directory: G:/code/sibras/vcpkg/buildtrees/wavpack/x64-windows-dbg
> See logs for more information:
> G:\code\sibras\vcpkg\buildtrees\wavpack\install-x64-windows-dbg-out.log
> 
> Call Stack (most recent call first):
> scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
> scripts/cmake/vcpkg_install_cmake.cmake:27 (vcpkg_build_cmake)
> ports/wavpack/portfile.cmake:25 (vcpkg_install_cmake)
> scripts/ports.cmake:128 (include)
> 
> Error: Building package wavpack:x64-windows failed with: BUILD_FAILED
> Please ensure you're using the latest portfiles with `.\vcpkg update`, then
> submit an issue at https://github.com/Microsoft/vcpkg/issues including:
> Package: wavpack:x64-windows
> Vcpkg version: 2021-01-13-unknownhash
> 
> [install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5894317/install-x64-windows-dbg-out.log)

This PR doesnt touch wavpack so thats an unrelated issue. Having a quick look at you log indicates that your Visual Studio installation is incorrect. You need to install the Visual C++ MFC package.

Sibras committed:
[libssh] Correct pthread naming on windows.

Sibras committed:
[fontconfig] Add libintl to pkgconfig on windows.

Sibras commented:
> 
> 
> When building `ffmpeg[ffmpeg,ffplay,ffprobe,avcodec,avformat,avdevice,avfilter,postproc,swresample,swscale,avresample,nonfree,gpl,version3,all-gpl,all-nonfree,bzip2,iconv,fdk-aac,fontconfig,freetype,fribidi,lzma,modplug,mp3lame,nvcodec,opencl,openjpeg,openssl,opus,sdl2,snappy,soxr,speex,ssh,tesseract,theora,vorbis,vpx,wavpack,webp,x264,xml2,zlib]:x64-windows-static`
> 
> ```
> === CONFIGURING ===
> ERROR: fontconfig not found using pkg-config
> 
> If you think configure made a mistake, make sure you are using the latest
> version from Git.  If the latest version fails, report the problem to the
> ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
> Include the log file "ffbuild/config.log" produced by configure as this will help
> solve the problem.
> ```
> 
> [config.log](https://github.com/microsoft/vcpkg/files/5877467/config.log)

Fixed in "[fontconfig] Add libintl to pkgconfig on windows."

cenit commented on ports/libssh/portfile.cmake:
```diff
@@ -40,6 +36,12 @@ vcpkg_configure_cmake(
 vcpkg_copy_pdbs()
+#Fixup pthread naming
+if(NOT VCPKG_TARGET_IS_MINGW AND VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/libssh.pc" "-lpthread" "-lpthreadVC3d")
```
So is it depending on pthread port?

Sibras replied:
libssh uses mbedtls which depends on pthread port

Sibras committed:
update version records

Sibras committed:
[ffmpeg] speex now supports non-windows.

Sibras committed:
[ffmpeg] Add feature libass.

Sibras committed:
[ffmpeg] Add dav1d feature.

Sibras committed:
[ffmpeg] Add feature ilbc.

Sibras committed:
[ffmpeg] Fixup FindFFmpeg.

Sibras committed:
[ffmpeg] Add tensorflow feature.

Sibras committed:
[ffmpeg] update version record.

Sibras commented:
This PR closes #13227, #13111

cenit commented:
> This PR closes #13227, #13111

Bonus point: if you add ("Fixes" or "Closes" work the same)
Closes: #13227
Closes: #13111
to the first message it will do it automatically (close the issues) when merged :)

strega-nil commented:
/azp run

cenit commented:
@Sibras to ease feature review, it might be useful if you create a vcpkg-ci-ffmpeg with as many features enabled as possible for each CI config (similar to vcpkg-ci-opencv for example). You need to create an empty portfile inside scripts/test_ports and make it depend using a CONTROL file on a combination of ffmpeg features (might even be dependent on ffmpeg[all] if you are lucky you can test it).

Of course, pay attention that the port is compiled and not skipped as a cascade failure, otherwise you might not notice that you self-excluded yourself to *all* CI tests without any further message/notice/regression (unfortunately, that's why you have to be careful when you set it up)

Sibras committed:
[ffmpeg] Add CI feature test.

cenit commented:
Please remove some features for these configs (use syntax exactly like vcpkg-ci-opencv for feature sets different for each config):
 
```
vcpkg-ci-ffmpeg:x64-uwp:    cascade
vcpkg-ci-ffmpeg:arm64-windows:    cascade
vcpkg-ci-ffmpeg:x64-uwp:    cascade
```

Others are still running, so there might be more.

Sibras commented:
> 
> 
> Please remove some features for these configs, or fix dependencies that cannot be built so that the port can run tests (use syntax exactly like vcpkg-ci-opencv for feature sets different for each config):
> 
> ```
> vcpkg-ci-ffmpeg:x64-uwp:    cascade
> vcpkg-ci-ffmpeg:arm64-windows:    cascade
> vcpkg-ci-ffmpeg:x64-uwp:    cascade
> ```
> 
> Others are still running, so there might be more.

Where can I see the failure logs for vcpkg-ci-ffmpeg? or do I have to run them locally?

Sibras committed:
[ffmpeg] Limit features based on CI failures.

Sibras committed:
[ffmpeg] Update version record.

cenit commented:
ok these triplets are not testing ffmpeg (and all depending ports) because of the cascade failure, so please consider also the following in addition to the normal CI results (otherwise you would turn off CI for many ports, opencv for example, in the respective triplets)

```
vcpkg-ci-ffmpeg:x64-windows-static-md:    cascade
vcpkg-ci-ffmpeg:x64-uwp:    cascade
vcpkg-ci-ffmpeg:arm64-windows:    cascade
vcpkg-ci-ffmpeg:arm-uwp:    cascade
```

Sibras commented:
> 
> 
> ok these triplets are not testing ffmpeg (and all depending ports) because of the cascade failure, so please consider also the following in addition to the normal CI results (otherwise you would turn off CI for many ports, opencv for example, in the respective triplets)
> 
> ```
> vcpkg-ci-ffmpeg:x64-windows-static-md:    cascade
> vcpkg-ci-ffmpeg:x64-uwp:    cascade
> vcpkg-ci-ffmpeg:arm64-windows:    cascade
> vcpkg-ci-ffmpeg:arm-uwp:    cascade
> ```

Is there anyway to get the ffbuild/config.log for the failing builds from the CI as i cant test/reproduce some of the errors.

JackBoosY commented:
@Sibras  I will test it manually then provide the failure log to you.

Sibras committed:
[ffmpeg] limit features based on ci.baseline.

Sibras committed:
[various ports] Update supports field.

Sibras committed:
[ffmpeg] Limit features more based on CI.

Sibras committed:
update version records.

Sibras commented:
> 
> 
> @Sibras I will test it manually then provide the failure log to you.

Thanks, its mainly the linux/osx ones I cant test

Sibras committed:
[ffmpeg] Add detection of additional non target deps..

select_library_configurations_from_names currently detects the debug libs even for release builds as  _IMPORT_PREFIX was not being set.

Sibras committed:
[aubio] Silence warning about FindFFMPEG.

Sibras committed:
[pangolin] Use vcpkg supplied FindFFMPEG.

Sibras committed:
update version files.

JackBoosY commented:
x64-linux regression:
```
=== CONFIGURING ===
ERROR: libssh not found using pkg-config

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
```

Sibras commented:
> 
> 
> x64-linux regression:
> 
> ```
> === CONFIGURING ===
> ERROR: libssh not found using pkg-config
> 
> If you think configure made a mistake, make sure you are using the latest
> version from Git.  If the latest version fails, report the problem to the
> ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
> Include the log file "ffbuild/config.log" produced by configure as this will help
> solve the problem.
> ```

Do you have the "ffbuild/config.log" file as I need that to know why it failed. Trying compilation in wsl actually succeeds so I cant reproduce the issue locally

Sibras committed:
[ffnvcodec] Set as not supporting uwp.

ffnvcodec will build under uwp as its a header only lib, but it can not be used as it requires dynamic dll loading.

cenit commented:
WSL if working on /mnt folder does not reveal filesystem case issues. May that be the case?

Sibras commented:
> 
> 
> WSL if working on /mnt folder does not reveal filesystem case issues. May that be the case?

No idea without detailed logs. I suspect its a pthread issue not being pulled in correctly by libssh. But there must be a difference between the system libs on my wsl and the CI thats causing things to get picked up differently. If that is the case though I would need the config.log and preferably the libssh.pc from a failing build. Baring that I could potentially change libssh to use openssl instead of mbedtls (as mbedtls is the one that doesnt pull in pthread properly) and just see if that fixes the CI

JackBoosY commented:
@Sibras Will provide you the log later.

JackBoosY commented:
```
=== CONFIGURING ===
ERROR: libx265 not found

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
```

config.log:
```
gcc -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200112 -D_XOPEN_SOURCE=600 -DPIC -std=c11 -fomit-frame-pointer -fPIC -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/harfbuzz -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/fribidi -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/libpng16 -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -I/home/usr/work/15787/vcpkg/packages/dav1d_x64-linux/include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/libpng16 -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/uuid -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/libpng16 -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/fribidi -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/opus -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/opus -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -c -o /tmp/ffconf.x6HAse6S/test.o /tmp/ffconf.x6HAse6S/test.c
gcc -Wl,--as-needed -Wl,-z,noexecstack -pthread -o /tmp/ffconf.x6HAse6S/test /tmp/ffconf.x6HAse6S/test.o -lx265 -lm -ldl -lstdc++ -lgcc_s -lgcc -lrt -lnuma
/usr/bin/ld: cannot find -lnuma
collect2: error: ld returned 1 exit status
ERROR: libx265 not found
```

Sibras committed:
Merge branch 'master' into master

Sibras committed:
Merge branch 'master' of https://github.com/Sibras/vcpkg

Sibras committed:
[ffmpeg] Update feature all.

Sibras committed:
update version records

cenit commented on ports/tesseract/portfile.cmake:
```diff
@@ -47,6 +47,9 @@ file(WRITE ${CURRENT_PACKAGES_DIR}/share/tesseract/TesseractConfig.cmake "${TESS
 
 vcpkg_copy_tools(TOOL_NAMES tesseract AUTO_CLEAN)
 
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/tesseract.pc" "-ltesseract41" "-ltesseract41d")
```
this command is not going to work in a release-only triplet

cenit replied:
can you wrap it around a if(NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")?

Sibras replied:
done

Sibras committed:
[tesseract] Wrap debug pkgcfg update.

cenit commented:
x64-windows-static-md: cascade failure
arm64-windows: cascade failure
arm-uwp: cascade failure

Those are green flags but should be red

Sibras commented:
> 
> 
> x64-windows-static-md: cascade failure
> arm64-windows: cascade failure
> arm-uwp: cascade failure
> 
> Those are green flags but should be red

How do you get information about cascade failures?

cenit commented:
That's a little bit difficult but doable

x64-windows-static-md: go here https://dev.azure.com/vcpkg/public/_build/results?buildId=49368&view=logs&j=837527cd-1f23-573f-4465-1412393a9ad5&t=5ebe51be-b0b9-5501-ce62-e5d2bdd3ba19 and scroll the list until you find "vcpkg-ci-ffmpeg:x64-windows-static-md". You will find that it is not tested because of cascade failure.
Similarly for all other triplets, be sure that the port vcpkg-ci-ffmpeg:triplet-name is checked (marked with *) or already checked (marked with passed). 
If you find "cascade", it means that the port is dependent on another port which is broken in that triplet (or marked unsupported, which is a terribly wrong option imho for vcpkg but still because if it ever get fixed it would not be tested anyway): in this case, either you fix the broken dependency or you just make sure that you do not depend on it anymore

Sibras committed:
[libssh,libxml2,openjpeg,fontconfig] Fix pkg-config for release only triplets.

Sibras committed:
[libssh] Correct port version after merge.

Sibras committed:
[ffmpeg] Fixup after merge.

Sibras committed:
Update version files.

Sibras committed:
[ffmpeg] Add opengl support.

Sibras commented:
> 
> 
> That's a little bit difficult but doable
> 
> x64-windows-static-md: go here https://dev.azure.com/vcpkg/public/_build/results?buildId=49368&view=logs&j=837527cd-1f23-573f-4465-1412393a9ad5&t=5ebe51be-b0b9-5501-ce62-e5d2bdd3ba19 and scroll the list until you find "vcpkg-ci-ffmpeg:x64-windows-static-md". You will find that it is not tested because of cascade failure.
> Similarly for all other triplets, be sure that the port vcpkg-ci-ffmpeg:triplet-name will be checked (marked with *) or has already been checked (marked with passed).
> If you find "cascade", it means that the port is dependent on another port which is broken in that triplet (or marked unsupported, which is a terribly wrong option imho for vcpkg but still because if it ever get fixed it would not be tested anyway): in this case, either you fix the broken dependency or you just make sure that you do not depend on it anymore

thanks, I actually looked through that page but ctrl-f wasnt finding any instances of 'cascade'. Turns out I needed to open it as a raw file and then ctrl-f worked 🤷 

I cant fully test the arm fails as some of the issues I can reproduce on my machine are due to cross-compiling (some projects compile an executable and then run that in order to generate source files which doesn't work as the executable is the wrong arch for the host). But im not sure if its possible to only disable them when cross compiling. For the moment i will just disable them for arm completely unless you have a better solution

Sibras committed:
[ffmpeg] Update package version.

Sibras committed:
[ffmpeg] Fix ffnvcodec support.

Sibras committed:
[ffmpeg] Fix x265 detection on osx.

cenit commented:
> thanks, I actually looked through that page but ctrl-f wasnt finding any instances of 'cascade'. Turns out I needed to open it as a raw file and then ctrl-f worked 🤷 
> 

there is also a search field on the page. Search offered from the browser unfortunately does not usually work very well.

> 
> I cant fully test the arm fails as some of the issues I can reproduce on my machine are due to cross-compiling (some projects compile an executable and then run that in order to generate source files which doesn't work as the executable is the wrong arch for the host). But im not sure if its possible to only disable them when cross compiling. For the moment i will just disable them for arm completely unless you have a better solution

please consider that it's still very common for arm to be a cross-compiling target, not a native build. And this is valid in CI too (it's not a native arm machine doing the build)
So yes, please disable on that arch whatever might not work out of the box and leave it as an unsupported feature (and untested). 
ffmpeg was tested on all architectures and was fine before, so it would be a pity to completely loose all tests due to a "cascade" tag, because in this unfortunate case also tests on the ffmpeg[core] port are losts

Sibras commented:
> 
> 
> ```
> === CONFIGURING ===
> ERROR: libx265 not found
> 
> If you think configure made a mistake, make sure you are using the latest
> version from Git.  If the latest version fails, report the problem to the
> ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
> Include the log file "ffbuild/config.log" produced by configure as this will help
> solve the problem.
> ```
> 
> config.log:
> 
> ```
> gcc -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200112 -D_XOPEN_SOURCE=600 -DPIC -std=c11 -fomit-frame-pointer -fPIC -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/harfbuzz -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/fribidi -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/libpng16 -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -I/home/usr/work/15787/vcpkg/packages/dav1d_x64-linux/include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/libpng16 -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/uuid -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/libpng16 -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/fribidi -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/opus -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include/opus -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/packages/libvpx_x64-linux/include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -I/home/usr/work/15787/vcpkg/installed/x64-linux/lib/pkgconfig/../../include -pthread -c -o /tmp/ffconf.x6HAse6S/test.o /tmp/ffconf.x6HAse6S/test.c
> gcc -Wl,--as-needed -Wl,-z,noexecstack -pthread -o /tmp/ffconf.x6HAse6S/test /tmp/ffconf.x6HAse6S/test.o -lx265 -lm -ldl -lstdc++ -lgcc_s -lgcc -lrt -lnuma
> /usr/bin/ld: cannot find -lnuma
> collect2: error: ld returned 1 exit status
> ERROR: libx265 not found
> ```

Thanks, Ive fixed that error (I think) but x265 on osx still has issues and Id need the log to fix it. Also the osx x265.pc file would be useful aswell.

Im pretty sure its still this line giving issues:
> gcc -Wl,--as-needed -Wl,-z,noexecstack -pthread -o /tmp/ffconf.x6HAse6S/test /tmp/ffconf.x6HAse6S/test.o -lx265 -lm -ldl -lstdc++ -lgcc_s -lgcc -lrt

But im not sure what the correct dependency libs are for osx.

Sibras committed:
[libvpx] Enable arm-uwp build.

cenit commented:
all triplets are correctly tested. This is already a wonderful achievement.

I am running the x64-osx pipeline locally to be able to provide you all the logs that you might be interested in. First of all, I hope to reproduce correctly the same error

cenit commented:
```
=== CONFIGURING ===
ERROR: libx265 not found

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
```

[config.log](https://github.com/microsoft/vcpkg/files/5977605/config.log)

[x265.pc.txt](https://github.com/microsoft/vcpkg/files/5977610/x265.pc.txt)

cenit commented:
@Sibras let me know if you want me to try any test for you

Sibras committed:
[ffmpeg] Fixup x265 patch.

Sibras commented:
> 
> 
> @Sibras let me know if you want me to try any test for you

Thanks, that was enough to fix the x265 error. However there is a new one with sdl2 on osx. So again ill need the complete config.log and the sdl2.pc file to be able to work out whats going on.

cenit commented:
try triggering an sdl2 bump increasing port version. I cannot reproduce this error (but stil it's failing because of another OpenGL problem)

[build-x64-osx-rel-err.log](https://github.com/microsoft/vcpkg/files/5978236/build-x64-osx-rel-err.log)

Sibras committed:
trigger sdl rebuild

Sibras committed:
[ffmpeg] Disable opengl on osx.

Sibras commented:
> 
> 
> try triggering an sdl2 rebuild bumping the port version. I cannot reproduce this error (but stil it's failing because of another OpenGL problem)
> 
> [build-x64-osx-rel-err.log](https://github.com/microsoft/vcpkg/files/5978236/build-x64-osx-rel-err.log)

On investigation it appears there is a known compilation error with opengl when glx.h is installed. Since there is now way to guarantee that glx.h is not available i have just disabled opengl on osx.

A version bump of sdl2 didnt appear to fix the osx issue. So I have no idea whats causing the CI to fail. I might have to just disable it on osx until we can sort out whats the problem.

cenit commented:
with https://github.com/microsoft/vcpkg/pull/15787/commits/518f0db99f903a27494a43d0119b10f4f3dfd390 on my Mac this is what I get. It seems that the previous error is removed (as expected, now the feature is disabled), but dav1d is still not working (only in the debug config, the release one is ok)


build-x64-osx-dbg-out.log:

```
=== CONFIGURING ===
ERROR: dav1d >= 0.4.0 not found using pkg-config

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
```

[config.log](https://github.com/microsoft/vcpkg/files/5989795/config.log)

mcmtroffaes commented on ports/ffmpeg/FindFFMPEG.cmake.in:
```diff
@@ -82,22 +98,54 @@ if(UNIX AND NOT APPLE)
 
+if(@ENABLE_ASS@)
+  select_library_configurations_from_names(BASENAME ASS NAMES ass libass)
+  list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS ${ASS_LIBRARIES})
```
This is missing some dependencies of libass. Here's a build on windows and macos of a simple test program compiled against ffmpeg[core,ass,avfilter]:

https://ci.appveyor.com/project/mcmtroffaes/ffmpeg-msvc-build/build/job/vi2om05rsvsw91b2#L816
https://travis-ci.com/github/mcmtroffaes/ffmpeg-msvc-build/jobs/483418890#L582

See https://github.com/microsoft/vcpkg/blob/master/ports/libass/vcpkg.json for a list of dependencies. It would be nice if we could use select_library_configurations_from_targets instead to handle this automatically, but I'm not sure if a target is available for libass on vcpkg.

Sibras replied:
Should be fixed now

mcmtroffaes replied:
Thanks a lot! I'll rerun the tests, and let you know how it goes.

mcmtroffaes replied:
I'm pleased to confirm the fix. Thanks again!

Sibras committed:
Revert "trigger sdl rebuild"

This reverts commit 94065bfe8414259ad5a5576bd4ac0b7ab4b97c6b.

Sibras committed:
[ffmpeg] Disable failing features on osx.

mcmtroffaes commented on ports/ffmpeg/FindFFMPEG.cmake.in:
```diff
@@ -82,22 +98,54 @@ if(UNIX AND NOT APPLE)
 
+if(@ENABLE_DAV1D@)
+  select_library_configurations_from_names(BASENAME DAV1D NAMES dav1d libdav1d)
+  list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS ${DAV1D_LIBRARIES})
```
The dav1d libary additionally needs ``list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS dl)`` on linux. For instance, see:

https://travis-ci.com/github/mcmtroffaes/ffmpeg-msvc-build/jobs/483418933#L494

mcmtroffaes replied:
```suggestion
  list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS ${DAV1D_LIBRARIES})
  if(UNIX)
    list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS dl)
  endif()
```

Sibras committed:
Update ports/ffmpeg/FindFFMPEG.cmake.in

Co-authored-by: Matthias C. M. Troffaes <matthias.troffaes@gmail.com>

Sibras committed:
[ffmpeg] Add ass dependencies to FindFFmpeg.

mcmtroffaes commented on ports/ffmpeg/FindFFMPEG.cmake.in:
```diff
@@ -195,6 +291,12 @@ if(@ENABLE_X265@)
   list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS ${X265_LIBRARIES})
```
```suggestion
  list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS ${X265_LIBRARIES})
  if(UNIX)
    list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS dl)
  endif()
```

... see https://travis-ci.com/github/mcmtroffaes/ffmpeg-msvc-build/jobs/483737650#L407

mcmtroffaes commented:
Thanks again for the excellent work on this PR, Sibras. My first batch of runs have finally finished. It turned up a few linux related issues. None of these are blocking IMO, and I can help resolving these after the PR is merged.

* ffmpeg[core,freetype,avcodec] - library dependency resolution issue on linux, looks like an issue in the FindFFMPEG.cmake script but haven't found out yet what exactly is triggering it: https://travis-ci.com/github/mcmtroffaes/ffmpeg-msvc-build/jobs/483737620#L397
* ffmpeg[core,ssh,avformat] - it appears that pkg-config does not find libssh, haven't looked into details yet: https://travis-ci.com/github/mcmtroffaes/ffmpeg-msvc-build/jobs/483737634#L456

It also brought up some issues unrelated to ffmpeg:

* fontconfig - broken on linux when gperf is missing (not sure if vcpkg is meant to install that by itself or not, easy fix by adding it to the CI environment, but I thought I'd flag it here just in case): https://travis-ci.com/github/mcmtroffaes/ffmpeg-msvc-build/jobs/483737621#L656
* pthreads (a dependency of ffmpeg[core,freetype,fontconfig,fribidi,avfilter]) - on the x64-windows-static-md triplet, pthreads has the wrong CRT linkage: https://ci.appveyor.com/project/mcmtroffaes/ffmpeg-msvc-build/builds/37815480/job/15jwnhohb2i8olfq#L104

JackBoosY commented:
/azp run

JackBoosY commented:
gdal regression is not related to this PR.

Sibras committed:
Update ports/ffmpeg/FindFFMPEG.cmake.in

Co-authored-by: Matthias C. M. Troffaes <matthias.troffaes@gmail.com>

JackBoosY commented:
Please run `./vcpkg x-add-version --overwrite-version ffmpeg` and commit changes.

Sibras committed:
update version

JackBoosY commented:
I will retest all features manually later.

cenit commented:
> I will retest all features manually later.

@JackBoosY we set up vcpkg-ci-ffmpeg exactly to help you. Please do not re-test configs that are known as broken

HppZ commented:
`vcpkg install ffmpeg[ffmpeg,openssl,gpl,xml2,x264,vorbis,vpx] --triplet=x86-uwp
`
Couldn't find this tool:
opensource/vcpkg/packages/ffmpeg_x86-uwp/bin/ffmpeg.exe.

Sibras commented:
> 
> 
> `vcpkg install ffmpeg[ffmpeg,openssl,gpl,xml2,x264,vorbis,vpx] --triplet=x86-uwp `
> Couldn't find this tool:
> opensource/vcpkg/packages/ffmpeg_x86-uwp/bin/ffmpeg.exe.

The tools arent supported in uwp builds

HppZ commented:
yes, so I deleted thoese `vcpkg_copy_tools` calls and I am rebuilding now.

HppZ commented:
building succeed. but my app can't launch. https://github.com/microsoft/vcpkg/issues/16529

Exception thrown: 'System.IO.FileLoadException' in AppName.exe
Activation of the Windows Store app '' failed with error 'Operation not supported. Unknown error: 0x80040904'.
![image](https://user-images.githubusercontent.com/12285582/110197888-71374900-7e89-11eb-8e23-7d22412281fc.png)
![image](https://user-images.githubusercontent.com/12285582/110197900-798f8400-7e89-11eb-9e2c-d0c5cdb5ba9f.png)

cenit commented:
> > 
> 
> > 
> 
> > `vcpkg install ffmpeg[ffmpeg,openssl,gpl,xml2,x264,vorbis,vpx] --triplet=x86-uwp `
> 
> > Couldn't find this tool:
> 
> > opensource/vcpkg/packages/ffmpeg_x86-uwp/bin/ffmpeg.exe.
> 
> 
> 
> The tools arent supported in uwp builds

wouldn't it be better to force-disable tools in uwp and emit a warning to the user that the feature request was not obeyed, instead of erroring out?

Sibras commented:
> 
> 
> > > 
> > 
> > 
> > > 
> > 
> > 
> > > `vcpkg install ffmpeg[ffmpeg,openssl,gpl,xml2,x264,vorbis,vpx] --triplet=x86-uwp `
> > 
> > 
> > > Couldn't find this tool:
> > 
> > 
> > > opensource/vcpkg/packages/ffmpeg_x86-uwp/bin/ffmpeg.exe.
> > 
> > 
> > The tools arent supported in uwp builds
> 
> wouldn't it be better to force-disable tools in uwp and emit a warning to the user that the feature request was not obeyed, instead of erroring out?

Probably, although that is unrelated to this PR so that probably should be a separate PR after this has been merged

HppZ commented:
does this PR support uwp?

cenit commented:
> 
> 
> Probably, although that is unrelated to this PR so that probably should be a separate PR after this has been merged

you're partially right: why is it passing in ci, if a reduced set is broken for users? that's my only concern related to this PR

HppZ commented:
> why is it passing in ci, if a reduced set is broken for users? 

I agree. Actually there are more broken ports for uwp, which make me feel sad and confusing.

HppZ commented:
please care more on UWP.

Sibras commented:
> 
> 
> > Probably, although that is unrelated to this PR so that probably should be a separate PR after this has been merged
> 
> you're partially right: why is it passing in ci, if a reduced set is broken for users? that's my only concern related to this PR

This PR just added some additional library optional dependencies and tested that those work. The issue with the tools not being built with uwp was pre-existing and this PR doesnt touch the tools hence why i suggested it should be a separate PR/Issue. Ive already got a potential fix for the tools but seeing as how this PR has been pending for a while I didnt want to complicate things further and potentially slow down it being merged. Once this PR has been applied ill create an additional PR that corrects the tools and adds those to the CI aswell.

Sibras committed:
[fontconfig] disable pthread/json as they are not needed for lib builds.

Only used for tests.

Sibras committed:
[ffmpeg] Enable fontconfig on static+windows.

Sibras committed:
Merge remote-tracking branch 'upstream/master'

Sibras committed:
update versions

Sibras committed:
update versions.

Sibras committed:
Fix incorrectly included commits

JackBoosY commented:
When building `pangolin:x64-linux`:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
SPEEX_LIBRARY
    linked by target "pangolin" in directory /mnt/vcpkg-ci/buildtrees/pangolin/src/v0.5-623ba41d18.clean/src
_dep_dbg
    linked by target "pangolin" in directory /mnt/vcpkg-ci/buildtrees/pangolin/src/v0.5-623ba41d18.clean/src
    linked by target "pangolin" in directory /mnt/vcpkg-ci/buildtrees/pangolin/src/v0.5-623ba41d18.clean/src

CMake Generate step failed.  Build files cannot be regenerated correctly.
```

Sibras committed:
revert pangolin commit

Reverts most of 2543be2edf60bf38511c2d477bb17b617398a108

JackBoosY commented:
@Sibras Do you need to add other changes?

Sibras commented:
> 
> 
> @Sibras Do you need to add other changes?

Nope, should be all good. I just had to fix some things due to merge conflicts.

JackBoosY commented:
@Sibras  Sorry but please resolve the merge conflicts.

Thanks.

Sibras committed:
Merge remote-tracking branch 'upstream/master'

Sibras committed:
update versions

Sibras commented:
> 
> 
> @Sibras Sorry but please resolve the merge conflicts.
> 
> Thanks.

@JackBoosY Done

Sibras committed:
Merge remote-tracking branch 'upstream/master'

Sibras committed:
Correct port version after merge.

JackBoosY commented:
Please do not incorporate other modifications into this PR.

Sibras committed:
Merge remote-tracking branch 'upstream/master'

JackBoosY commented:
Waiting for merge #16904.

JackBoosY commented:
Can you please solve the file conflicts?

Thanks,

Sibras committed:
Merge branch 'master' into master

Sibras committed:
update versions.

Sibras committed:
[ffmpeg] Fix cmake dependency detection on non-windows.

Sibras committed:
Revert "revert pangolin commit"

This reverts commit f59bc5a53ea74fd9df17fc368886e9d1a80ac42f.

Sibras committed:
[ffmpeg] Fix dependency loading that does not define separate debug/release libs.

Sibras committed:
update versions.

Sibras committed:
[ffmpeg] set CMP0072 policy.

Sibras committed:
[wavpack] Fix cmake config export.

JackBoosY commented:
When building `pangolin:x64-linux`:
```
[66/77] : && /usr/bin/c++ -std=c++0x -Wall -Wextra -fPIC -g  examples/HelloPangolin/CMakeFiles/HelloPangolin.dir/main.cpp.o -o examples/HelloPangolin/HelloPangolin  src/libpangolin.a  -lrt  -lpthread  -pthread  -Wl,--no-undefined  /usr/lib/x86_64-linux-gnu/libGL.so  /usr/lib/x86_64-linux-gnu/libGLU.so  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLEWd.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  /usr/lib/x86_64-linux-gnu/libpython3.6m.so  -ldl  -lutil  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavdevice.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavfilter.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavformat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavcodec.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpostproc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswscale.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavutil.a  -lX11  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libass.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libharfbuzz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbz2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libdav1d.a  -lc  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libilbc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfdk-aac.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfontconfig.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfreetyped.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libexpat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libuuid.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpng16d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlidec-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlicommon-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfribidi.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/liblzmad.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmp3lame.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmodplug.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libOpenCL.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopenjp2.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libssl.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libcrypto.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopus.a  -lm  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libSDL2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsnappyd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsoxr.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libspeex.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtheoraenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libogg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbis.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbisenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvpx.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwavpack.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdemuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpmuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdecoderd.a  -lpthread  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx264.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libjpeg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtiffd.a  /usr/lib/x86_64-linux-gnu/libGL.so  /usr/lib/x86_64-linux-gnu/libGLU.so  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLEWd.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  /usr/lib/x86_64-linux-gnu/libpython3.6m.so  -ldl  -lutil  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavdevice.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavfilter.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavformat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavcodec.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpostproc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswscale.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavutil.a  -lX11  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libass.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libharfbuzz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbz2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libdav1d.a  -lc  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libilbc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfdk-aac.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfontconfig.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfreetyped.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libexpat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libuuid.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpng16d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlidec-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlicommon-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfribidi.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/liblzmad.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmp3lame.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmodplug.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libOpenCL.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopenjp2.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libssl.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libcrypto.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopus.a  -lm  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libSDL2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsnappyd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsoxr.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libspeex.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtheoraenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libogg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbis.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbisenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvpx.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwavpack.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdemuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpmuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdecoderd.a  -lpthread  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx264.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libjpeg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtiffd.a && :
FAILED: examples/HelloPangolin/HelloPangolin 
: && /usr/bin/c++ -std=c++0x -Wall -Wextra -fPIC -g  examples/HelloPangolin/CMakeFiles/HelloPangolin.dir/main.cpp.o -o examples/HelloPangolin/HelloPangolin  src/libpangolin.a  -lrt  -lpthread  -pthread  -Wl,--no-undefined  /usr/lib/x86_64-linux-gnu/libGL.so  /usr/lib/x86_64-linux-gnu/libGLU.so  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLEWd.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  /usr/lib/x86_64-linux-gnu/libpython3.6m.so  -ldl  -lutil  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavdevice.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavfilter.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavformat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavcodec.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpostproc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswscale.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavutil.a  -lX11  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libass.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libharfbuzz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbz2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libdav1d.a  -lc  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libilbc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfdk-aac.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfontconfig.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfreetyped.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libexpat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libuuid.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpng16d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlidec-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlicommon-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfribidi.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/liblzmad.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmp3lame.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmodplug.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libOpenCL.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopenjp2.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libssl.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libcrypto.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopus.a  -lm  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libSDL2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsnappyd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsoxr.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libspeex.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtheoraenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libogg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbis.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbisenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvpx.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwavpack.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdemuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpmuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdecoderd.a  -lpthread  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx264.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libjpeg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtiffd.a  /usr/lib/x86_64-linux-gnu/libGL.so  /usr/lib/x86_64-linux-gnu/libGLU.so  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLEWd.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  /usr/lib/x86_64-linux-gnu/libpython3.6m.so  -ldl  -lutil  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavdevice.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavfilter.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavformat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavcodec.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpostproc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswresample.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libswscale.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libavutil.a  -lX11  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libass.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libharfbuzz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbz2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libdav1d.a  -lc  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libilbc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfdk-aac.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfontconfig.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfreetyped.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libexpat.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libuuid.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libz.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libpng16d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlidec-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libbrotlicommon-static.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libfribidi.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/liblzmad.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmp3lame.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libmodplug.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libOpenCL.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopenjp2.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libssl.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libcrypto.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libopus.a  -lm  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libSDL2d.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsnappyd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libsoxr.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libspeex.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtheoraenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libogg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbis.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvorbisenc.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libvpx.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwavpack.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdemuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpmuxd.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libwebpdecoderd.a  -lpthread  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx264.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libjpeg.a  /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libtiffd.a && :
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a(threadpool.cpp.o): In function `numa_free_nodemask':
/usr/include/numa.h:183: undefined reference to `numa_bitmask_free'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a(threadpool.cpp.o): In function `numa_free_cpumask':
/usr/include/numa.h:279: undefined reference to `numa_bitmask_free'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a(threadpool.cpp.o): In function `x265::ThreadPool::allocThreadPools(x265_param*, int&, bool)':
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:268: undefined reference to `numa_available'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:283: undefined reference to `numa_allocate_cpumask'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:286: undefined reference to `numa_node_to_cpus'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:288: undefined reference to `numa_bitmask_weight'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a(threadpool.cpp.o): In function `x265::ThreadPool::create(int, int, unsigned long)':
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:489: undefined reference to `numa_available'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:491: undefined reference to `numa_allocate_nodemask'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a(threadpool.cpp.o): In function `x265::ThreadPool::setThreadNodeAffinity(void*)':
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:583: undefined reference to `numa_available'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:585: undefined reference to `numa_run_on_node_mask'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:586: undefined reference to `numa_set_interleave_mask'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:587: undefined reference to `numa_set_localalloc'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libx265.a(threadpool.cpp.o): In function `x265::ThreadPool::getNumaNodeCount()':
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:606: undefined reference to `numa_available'
/mnt/vcpkg-ci/buildtrees/x265/src/f1b45711b7-44018d5a0c.clean/source/common/threadpool.cpp:607: undefined reference to `numa_max_node'
collect2: error: ld returned 1 exit status
```

Sibras committed:
[ffmpeg] Add optional system dependent libraries.

Sibras committed:
Merge branch 'master' into master

Sibras committed:
update versions after merge.

PhoebeHui commented:
We have same issue https://github.com/microsoft/vcpkg/issues/16990 reported about gdal:x64-osx, however, I haven't repro this issue locally.

```
...
./lib/autoconf/general.m4:2499: AC_CHECKING is expanded from...
configure.ac:4448: the top level
configure.ac:5611: warning: The macro `AC_CHECKING' is obsolete.
configure.ac:5611: You should run autoupdate.
./lib/autoconf/general.m4:2499: AC_CHECKING is expanded from...
configure.ac:5611: the top level
configure.ac:6009: warning: AC_OUTPUT should be used without arguments.
configure.ac:6009: You should run autoupdate.
autoreconf: running: /usr/local/Cellar/autoconf/2.71/bin/autoheader --force
autoreconf: configure.ac: not using Automake
autoreconf: './config.sub' is updated
autoreconf: './config.guess' is updated
configure.ac: error: required file 'config.rpath' not found
autoreconf: Leaving directory '.'


```

JackBoosY commented:
Can you please solve the file conflicts again?

Thanks,

Sibras committed:
Merge remote-tracking branch 'upstream/master'

Sibras committed:
update versions.

JackBoosY commented:
I will ensure that this PR is merged as soon as possible.

JackBoosY commented:
@vicroms @strega-nil Please merge this PR first.

strega-nil commented:
Thanks so much @Sibras :)
